esphome/components/lvgl/btnmatrix.py

```diff
@@ -0,0 +1,275 @@
+from esphome import automation
+import esphome.codegen as cg
+from esphome.components.key_provider import KeyProvider
+import esphome.config_validation as cv
+from esphome.const import CONF_ID, CONF_WIDTH
+from esphome.cpp_generator import MockObj
+
+from .automation import action_to_code
+from .btn import lv_btn_t
+from .defines import (
+    BTNMATRIX_CTRLS,
+    CONF_BUTTONS,
+    CONF_CONTROL,
+    CONF_ITEMS,
+    CONF_KEY_CODE,
+    CONF_MAIN,
+    CONF_ONE_CHECKED,
+    CONF_ROWS,
+    CONF_SELECTED,
+    CONF_TEXT,
+)
+from .helpers import lvgl_components_required
+from .lv_validation import key_code, lv_bool
+from .lvcode import lv, lv_add, lv_expr
+from .schemas import automation_schema
+from .types import (
+    LV_BTNMATRIX_CTRL,
+    LV_STATE,
+    LvBoolean,
+    LvCompound,
+    LvType,
+    ObjUpdateAction,
+    char_ptr,
+    lv_pseudo_button_t,
+)
+from .widget import Widget, WidgetType, get_widgets, widget_map
+
+CONF_BUTTONMATRIX = "buttonmatrix"
```

Comment:
I feel like since this is two words, it should be separated as such in config
```suggestion
CONF_BUTTON_MATRIX = "button_matrix"
```

Reply:
The LVGL name for that widget is "buttonmatrix" (was btnmatrix in 8.4, 9.x renamed it to buttonmatrix). I think it's best to keep consistency with the underlying implementation to avoid confusion.